### PR TITLE
fix: `print-paths` changed to `setup-file` in Lean v4.4.0

### PR DESCRIPTION
**Pull request by utensil**
## Description

[`LeanInk/Analysis/LeanContext.lean`](https://github.com/leanprover/LeanInk/blob/main/LeanInk/Analysis/LeanContext.lean) relies on `print-paths` to `initializeLakeContext`, but in leanprover/lean4#2858 which is part of Lean v4.4.0 release, `print-paths` is changed to `setup-file`, with an extra parameter and changed JSON output.

This PR bumps LeanInk to use [v4.4.0-rc1](https://github.com/leanprover/LeanInk/commit/b707d2b925804c92e6eb26a1f11eae312cdd729f), replacing `print-paths` with  `setup-file` and handles the parameter and output change accordingly.

There is probably no immediate need to merge this PR as doc-gen4 is using https://github.com/hargoniX/LeanInk/ which is 30+ commits behind `leanprover/LeanInk` and doesn't have this file, and I'm not aware of other major users.

More discussions on [Zulip](https://leanprover.zulipchat.com/#narrow/stream/270676-lean4/topic/LeanInk.3A.20unknown.20command.20'print-paths').

### CODE REVIEW

**Comment by utensil** [comment]
There is no way I can pass the test for LeanInk. I tried the 2 approaches:

1. bumping Lean for LeanInk, so LeanInk itself and the target Lean program are both on v4.4.0, but tests in LeanInk are outdated by this leap of Lean toolchain, causing many failures that I don't intend to fix in this PR (or should I?). (this is now at my `fix-print-path-bump` branch)
2. don't bump Lean for LeanInk, so LeanInk itself is on old Lean, it works fine for a target Lean program on v4.4.0 (locally), but tests in LeanInk are still failing because now the target Lean programs are the tests in LeanInk on the old Lean toolchain.

**Comment by utensil** [comment]
I'll take approach 1 later. In the meanwhile, code in this PR can `lake build` and works for target Lean code on Lean v4.4.0.

File: test/bench/Basic.lean.leanInk.expected [hunk 1]
```diff
@@ -980,7 +980,7 @@
...
     "_type": "token"},
    {"typeinfo":
-    {"type": "{α : Type ?u.3247} → [self : BEq α] → α → α → Bool",
+    {"type": "{α : Type ?u.3298} → [self : BEq α] → α → α → Bool",
```
Comment (utensil) [inline test/bench/Basic.lean.leanInk.expected:983]: De Bruijn index change across versions seems to be normal.

File: test/bench/Basic.lean.leanInk.expected [hunk 2]
```diff
@@ -1939,7 +1939,7 @@
     "raw": "absurd",
     "link": null,
     "docstring":
-    "Anything follows from two contradictory hypotheses. Example:\n```\nexample (hp : p) (hnp : ¬p) : q := absurd hp hnp\n```\nFor more information: [Propositional Logic](https://leanprover.github.io/theorem_proving_in_lean4/propositions_and_proofs.html#propositional-logic)\n",
+    "Anything follows from two contradictory hypotheses. Example:\n```\nexample (hp : p) (hnp : ¬p) : q := absurd hp hnp\n```\nFor more information: [Propositional Logic](https://lean-lang.org/theorem_proving_in_lean4/propositions_and_proofs.html#propositional-logic)\n",
```
Comment (utensil) [inline test/bench/Basic.lean.leanInk.expected:1942]: This file is full of this url change. Here is the meaningful diff produced by

```
git diff 0227adbf7ee9b323efa0c47bbd00c8976b51e4a1 71b8365e08fb8b786cff628bbea38a2512370ca7 test/bench/Basic.lean.leanInk.expected|grep -E '^[-+] '|grep -v theorem_proving_in_lean4
```

```diff
-    {"type": "{α : Type ?u.3298} → [self : BEq α] → α → α → Bool",
+    {"type": "{α : Type ?u.3247} → [self : BEq α] → α → α → Bool",
-   {"typeinfo": {"type": "n + m = k + m", "name": "h", "_type": "typeinfo"},
+   {"typeinfo": {"type": "m + n = k + m", "name": "h", "_type": "typeinfo"},
-    "conclusion": "0 = zero * m + zero * k",
+    "conclusion": "0 = 0 + zero * k",
-  [{"name": "zero",
-    "hypotheses":
-    [{"type": "Nat", "names": ["m", "k"], "body": "", "_type": "hypothesis"}],
-    "conclusion": "0 = 0 + zero * k",
+  [{"name": "",
+    "hypotheses": [],
+    "conclusion": "Goals accomplished! 🐙",
-  [{"name": "",
-    "hypotheses": [],
-    "conclusion": "Goals accomplished! 🐙",
+  [{"name": "zero",
+    "hypotheses":
+    [{"type": "Nat", "names": ["m", "k"], "body": "", "_type": "hypothesis"}],
+    "conclusion": "0 = zero * m + zero * k",
-   {"typeinfo": {"type": "m * n = k * m", "name": "h", "_type": "typeinfo"},
+   {"typeinfo": {"type": "n * m = k * m", "name": "h", "_type": "typeinfo"},
-    "raw": " ",
-    "link": null,
-    "docstring":
-    "In induction alternative, which can have 1 or more cases on the left\nand `_`, `?_`, or a tactic sequence after the `=>`.\n",
-    "_type": "token"}],
-  "_type": "sentence"},
- {"messages":
-  [{"contents":
-    "Error: unsolved goals\ncase succ\nn : Nat\nih : 0 - n = 0\n⊢ pred 0 = 0",
-    "_type": "message"}],
-  "goals":
-  [{"name": "succ",
-    "hypotheses":
-    [{"type": "Nat", "names": ["n"], "body": "", "_type": "hypothesis"},
-     {"type": "0 - n = 0", "names": ["ih"], "body": "", "_type": "hypothesis"}],
-    "conclusion": "0 - succ n = 0",
-    "_type": "goal"}],
-  "contents":
-  [{"typeinfo": null,
-    "semanticType": null,
-    "raw": "=> ",
+    "raw": " => ",
- {"messages":
-  [{"contents":
-    "Error: unsolved goals\ncase succ\nn : Nat\nih : 0 - n = 0\n⊢ pred 0 = 0",
-    "_type": "message"}],
+ {"messages": [],
-  [{"name": "succ",
-    "hypotheses":
-    [{"type": "Nat", "names": ["n"], "body": "", "_type": "hypothesis"},
-     {"type": "0 - n = 0", "names": ["ih"], "body": "", "_type": "hypothesis"}],
-    "conclusion": "pred 0 = 0",
+  [{"name": "",
+    "hypotheses": [],
+    "conclusion": "Goals accomplished! 🐙",
```

File: test/dep/Main.lean.leanInk.expected [hunk 3]
```diff
@@ -40,7 +40,7 @@
...
+    "The unit type, the canonical type with one element, named `unit` or `()`.\nIn other words, it describes only a single value, which consists of said constructor applied\nto no arguments whatsoever.\nThe `Unit` type is similar to `void` in languages derived from C.\n\n`Unit` is actually defined as `PUnit.{1}` where `PUnit` is the universe\npolymorphic version. The `Unit` should be preferred over `PUnit` where possible to avoid\nunnecessary universe parameters.\n\nIn functional programming, `Unit` is the return type of things that \"return\nnothing\", since a type with one element conveys no additional information.\nWhen programming with monads, the type `m Unit` represents an action that has\nsome side effects but does not return a value, while `m α` would be an action\nthat has side effects and returns a value of type `α`.\n",
```
Comment (utensil) [inline test/dep/Main.lean.leanInk.expected:43]: `PUnit.{0}` changed to ``PUnit.{1}`.

File: test/playground/infoTree.lean.leanInk.expected [hunk 4]
```diff
@@ -103,7 +103,7 @@
     "link": null,
     "docstring": null,
     "_type": "token"},
-   {"typeinfo": {"type": "A × A → B", "name": "B", "_type": "typeinfo"},
+   {"typeinfo": {"type": "Type", "name": "B", "_type": "typeinfo"},
```
Comment (utensil) [inline test/playground/infoTree.lean.leanInk.expected:106]: No idea why this has less information.

File: test/bench/Basic.lean.leanInk.expected [hunk 5]
```diff
@@ -33973,17 +33973,42 @@
...
   "_type": "sentence"},
- {"messages": [],
+ {"messages":
+  [{"contents":
+    "Error: unsolved goals\ncase succ\nn : Nat\nih : 0 - n = 0\n⊢ pred 0 = 0",
+    "_type": "message"}],
   "goals":
-  [{"name": "",
-    "hypotheses": [],
-    "conclusion": "Goals accomplished! 🐙",
```
Comment (utensil) [inline test/bench/Basic.lean.leanInk.expected:33986]: I miss this 🐙

File: test/bench/Basic.lean.leanInk.expected [hunk 6]
```diff
@@ -33973,17 +33973,42 @@
...
     "link": null,
     "docstring":
     "In induction alternative, which can have 1 or more cases on the left\nand `_`, `?_`, or a tactic sequence after the `=>`.\n",
     "_type": "token"}],
   "_type": "sentence"},
- {"messages": [],
+ {"messages":
+  [{"contents":
+    "Error: unsolved goals\ncase succ\nn : Nat\nih : 0 - n = 0\n⊢ pred 0 = 0",
+    "_type": "message"}],
```
Comment (utensil) [inline test/bench/Basic.lean.leanInk.expected:33985]: This proof is no longer proven, will investigate.

File: LeanInk/Analysis/LeanContext.lean [hunk 7]
```diff
@@ -51,11 +56,13 @@ def initializeLakeContext (lakeFile : FilePath) (header : Syntax) : AnalysisM Un
     | 0 => do
       let stdout := stdout.split (· == '\n') |>.getLast!
       match Json.parse stdout with
-      | Except.error _ => throw <| IO.userError s!"Failed to parse lake output: {stdout}"
+      | Except.error msg => throw <| IO.userError s!"Failed to parse lake output: {stdout}\nerror: {msg}"
```
Comment (utensil) [inline LeanInk/Analysis/LeanContext.lean:59]: The `msg` is helpful for debugging.

File: test/bench/Basic.lean.leanInk.expected (hunk 6 above)
Comment (utensil) [reply test/bench/Basic.lean.leanInk.expected:33985]:
```
@[simp] protected theorem zero_sub (n : Nat) : 0 - n = 0 := by
  induction n with
  | zero => rfl
  | succ n ih => simp [ih, Nat.sub_succ]
```

This proof failed to use the protected simp lemma `pred_zero` in the same namespace, is this expected?

**Comment by femtomc** [comment]
Note that this branch (and its fixes) also fixes my issue #56.

**Review by mhuisi — APPROVED** [review]

**Pull request merged**